apps/client/src/components/PokeCard/PokeCard.tsx

```diff
@@ -1,9 +1,39 @@
 import React from 'react';
-import { BasicPokemon } from '../../types/pokemons';
+import { FavoriteButton } from './FavoriteButton';
+import { useUpdatePokemonCache } from '../../hooks/useUpdatePokemonCache';
+import { useFavoriteApiOperations } from '../../hooks/useFavoriteApiOperations';
+import { isNullOrUndefined } from '../../utils';
+import toast from 'react-hot-toast';
+import { useSession } from '../../hooks/auth/useSession';
+import { Pokemon } from '@repo/shared-types';
+
+export const PokeCard: React.FC<Pokemon> = React.memo(function PokeCard({ id, name, isFavorite }) {
+  const { data: session } = useSession();
+  const updatePokemonCache = useUpdatePokemonCache();
+  const { removeFavorite, addFavorite } = useFavoriteApiOperations();
+
+  const onFavoriteClick = () => {
+    if (isNullOrUndefined(session) || !session.loggedIn) {
+      toast.error('Please login to favorite a pokemon');
+      return;
+    }
+
+    updatePokemonCache(id);
+
+    if (isFavorite) {
+      removeFavorite(id);
+    } else {
+      addFavorite(id);
+    }
```

Comment:
try to avoid using the `else` keyword

```suggestion
    if (isFavorite) {
      return removeFavorite(id);
    }
     
     addFavorite(id);
```